test/helpers/azuremonitor/mockoonServer.go

```diff
@@ -0,0 +1,140 @@
+package azuremonitor
+
+import (
+	_ "embed"
+	"time"
+
+	openshiftv1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	"github.com/openshift/cluster-logging-operator/test/framework/functional"
+	v1 "k8s.io/api/core/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:embed azure-http-data-collector-api.json
+var AzureHttpDataCollectorApi string
+
+const (
+	image   = "quay.io/openshift-logging/mockoon-cli:6.2.0"
+	port    = 3000
+	Mockoon = "mockoon"
+)
+
+type MockoonLog struct {
+	App             string      `json:"app,omitempty"`
+	EnvironmentName string      `json:"environmentName,omitempty"`
+	EnvironmentUUID string      `json:"environmentUUID,omitempty"`
+	Level           string      `json:"level,omitempty"`
+	Message         string      `json:"message,omitempty"`
+	RequestMethod   string      `json:"requestMethod,omitempty"`
+	RequestPath     string      `json:"requestPath,omitempty"`
+	RequestProxied  bool        `json:"requestProxied,omitempty"`
+	ResponseStatus  int         `json:"responseStatus,omitempty"`
+	Timestamp       time.Time   `json:"timestamp,omitempty"`
+	Transaction     Transaction `json:"transaction,omitempty"`
+}
+type Headers struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+type QueryParams struct {
+	APIVersion string `json:"api-version,omitempty"`
+}
+type Request struct {
+	Body        string      `json:"body,omitempty"`
+	Headers     []Headers   `json:"headers,omitempty"`
+	Method      string      `json:"method,omitempty"`
+	Params      []any       `json:"params,omitempty"`
+	Query       string      `json:"query,omitempty"`
+	QueryParams QueryParams `json:"queryParams,omitempty"`
+	Route       string      `json:"route,omitempty"`
+	URLPath     string      `json:"urlPath,omitempty"`
+}
+type Response struct {
+	Body          string    `json:"body,omitempty"`
+	Headers       []Headers `json:"headers,omitempty"`
+	StatusCode    int       `json:"statusCode,omitempty"`
+	StatusMessage string    `json:"statusMessage,omitempty"`
+}
+type Transaction struct {
+	Proxied           bool     `json:"proxied,omitempty"`
+	Request           Request  `json:"request,omitempty"`
+	Response          Response `json:"response,omitempty"`
+	RouteResponseUUID string   `json:"routeResponseUUID,omitempty"`
+	RouteUUID         string   `json:"routeUUID,omitempty"`
+}
+
+type MockoonServer struct {
+	configMap *v1.ConfigMap
+	service   *v1.Service
+	container *v1.Container
+	route     *openshiftv1.Route
+	Timeout   time.Duration
+}
+
+// NewMockoonServer creates a MockoonServer
+// Server will be created using Mockoon tools, specific environment described in azure-http-data-collector-api.json
+func NewMockoonServer(ns string, labels map[string]string) *MockoonServer {
+	ms := &MockoonServer{
+		service: runtime.NewService(ns, Mockoon),
```

Comment:
I think you will find moving these unnecessary by moving into container that runs within the context of the functional pod